frontend/src/javascript/utils/codeUtils.js

```diff
@@ -0,0 +1,95 @@
+import store from "@/store";
+
+export function compareCode(oldText, newText, event, selection) {
+    let oldLen = oldText.length;
+    let newLen = newText.length;
+    let lengthDifference = newLen - oldLen;
+
+    let oldLineCount = oldText.numberOfLines();
+    let newLineCount = newText.numberOfLines();
+    let lineCountDifference = newLineCount - oldLineCount;
+
+    let ret = {
+        start: selection.start,
+        end: selection.end,
+        size: lengthDifference,
+        deltaLineCount: lineCountDifference,
+    };
+
+    if (selection.start != selection.end) return ret;
+    if (lengthDifference >= 0) return ret;
+
+    if (!event.inputType.includes("Backward") && !event.inputType.includes("Forward")) {
+        console.error("Unhandled input type: ", event.inputType);
+        return ret;
+    }
+
+    if (event.inputType.includes("Backward")) {
+        ret.start -= Math.abs(lengthDifference);
+    } else {
+        ret.end += Math.abs(lengthDifference);
+    }
+
+    return ret;
+}
+
+export function moveTrackedVariables(changes) {
+    const handleVarTrackerMove = (ch, varObj) => {
+        if (varObj == null) return;
+
+        if (ch.end > varObj.start && ch.start < varObj.end) {
```

Comment:
> Łukasz: a co to za polityka kodu bez komentarzy? XD
> Marcin: Pisząc odpowiednio małe funkcje z opisowymi nazwami komentarze są zbędne, a zazwyczaj nawet nadmiarowe

No właśnie sztuką jest odpowiednie pisanie kodu, żeby sam siebie dokumentował. Ja wiem co w tym ifach się dzieje, bo sam Ci wcześniej podałem warunek na to, ale ktoś inny jak na to spojrzy będzie musiał chwilę pokminić, żeby dowiedzieć się, co ten warunek oznacza. Wyjścia są 2:
- Dodanie komentarza co w tych warunkach się liczy
- Przenieść ten warunek do np funkcji areIntervalsIntersect(interval1, interval2) w pliku intervalsUtils i wtedy nazwa funkcji będzie jednocześniej jej komentarzem (zalecany sposób)